src/Storages/AlterCommands.cpp

```diff
@@ -529,9 +530,95 @@ void AlterCommand::apply(StorageInMemoryMetadata & metadata, ContextPtr context)
         {
             metadata.columns.add(column, after_column, first);
         }
+
+        //Add secondary index for new column if enable_minmax_index_for_all_numeric_columns is enabled
+        if (isNumber(column.type) && metadata.settings_changes
```

Comment:
L. 535ff and l. 558ff are almost the same, let's merge them.

Reply:
addressed it